### ISSUE TITLE
Support for extracting CSS

### ISSUE THREAD

**Issue by Dwlad90** [microsoft/griffel#53]
I'd like to ask are you going to add support for extracting CSS?

**Comment by layershifter** [comment]
Hey, in short: **yes** 😸 

The long answer: yes, but it's something complicated 🤔 I will provide more technical details once will have time, but it's more or less the same problem as in https://github.com/atlassian-labs/compiled/issues/722.

**Comment by Dwlad90** [comment]
Thanks for the answer!
I’d glad when it happened. 👍🏻

**Comment by Dwlad90** [comment]
Hi @layershifter,
I guess 300Kb file isn’t possible in case with the atomic CSS, that’s why I think it’d be perfect to add optional extracting of CSS. It would give the opportunity to process styles with post processors and browse caching, so the styles loading only at first visit the page. If you have another way to work with post processors I would like to know.

**Comment by Dwlad90** [comment]
Thanks a lot!
I saw your pull request and I’m testing it now in my project. I’ll keep you updated.

**Comment by Dwlad90** [comment]
Thanks a lot, I really appreciate your help.

I successfully connected the package with the feature.

Extracting CSS from components works fine, but extracting styles from the page doesn’t work.


Print out the argument sourceCode of the function shouldTransformSourceCode in the file griffelWebpackExtractionLoader.ts:74:

```
import { makeStyles } from '@griffel/react';

const useClassNames = makeStyles({
  page: {
    color: '#768494',
  },
});

export function ContactUsPage() {
  const classNames = useClassNames();

  return <div className={classNames.page}>ContactUsPage</div>;
}

export default ContactUsPage;
```

Used packages:
"@nrwl/next": "13.8.6",
"next": "^12.1.0",
"react": "^17.0.2”

If you need more details I’d glad to help.

**Comment by layershifter** [comment]
@Dwlad90 this might be a problem with config (that is totally unclear without docs 🙄 ), I made a configured repo with latest Next.js: https://github.com/layershifter/griffel-extraction-test

![image](https://user-images.githubusercontent.com/14183168/159571259-44458be5-1586-4628-8cb9-83e5db715127.png)

![image](https://user-images.githubusercontent.com/14183168/159571310-3518fad3-ae76-478b-a9f9-87ede4f1e9a0.png)

**Comment by Dwlad90** [comment]
Thanks a lot, great job!
The bug reason was in the order of webpack loaders. I set order like in the example and now everything works perfect!

**Comment by layershifter** [comment]
Implemented for Webpack, check [`@griffel/webpack-extraction-plugin`](https://github.com/microsoft/griffel/tree/main/packages/webpack-extraction-plugin).

**Comment by Dwlad90** [comment]
> Implemented for Webpack, check [`@griffel/webpack-extraction-plugin`](https://github.com/microsoft/griffel/tree/main/packages/webpack-extraction-plugin).

Thanks a lot for a package, its a great job!

I was trying to use this package with nextJS using the mini-css-extract-plugin built-in but in its construction I get the following error:

`Global CSS cannot be imported from files other than your Custom <App>. Due to the Global natureof stylesheets, and to avoid conflicts, Please move all first-party global CSS imports to pages/_app.js. Or convert the import to Component-Level CSS (CSS Modules)`.

Sandbox: [](https://codesandbox.io/s/withered-firefly-2ff6m8)

I have to add the plugin with certain parameters to the pipeline so it’ll work properly:

`new MiniCssExtractPlugin({
    // without these Next.js will look for the generated stylesheets from the wrong place
    filename: "static/chunks/[chunkhash].css",
    chunkFilename: "static/chunks/[chunkhash].css",
    ignoreOrder: true,
  })`


Sandbox: [](https://codesandbox.io/s/hardcore-mendel-9xywyx)

Maybe you have an idea how can I use a built-in plugin?

**Comment by layershifter** [comment]
@Dwlad90 TBH I haven't tested it with Next.js, but I think that we can handle it. Would you mind to create a separate issue for this problem?

**Comment by pksorensen** [comment]
@layershifter - i am experimenting with css extraction. First attempt shows that it works for makeStyles, but the makeStaticStyles seems to not be extracted, atleast javascript have to run before those styles are applied. Trying to figure out why

**Comment by layershifter** [comment]
@pksorensen indeed, `makeStaticStyles()` is not supported for AOT/CSS extraction. We don't use it in products, so it was a lot priority for us. Can you please an issue?
